aarch64/src/kernel.ld

```diff
@@ -6,9 +6,7 @@ ENTRY(start)
 
 SECTIONS {
 	/*
-	 * start the kernel at 0x0xffff_8000_0020_0000
-	 * This preserves some RAM between 1MiB and the
-	 * start of the kernel for critical structures.
+	 * Raspberry Pi start address is 0x80_0000.
```

Comment:
Whoa. It really, really shouldn't be. I'd leave the start address here the same as it is on x86.

It's true right now we're running down at low addresses, but that's not tenable long-term.

Reply:
https://www.raspberrypi.com/documentation/computers/config_txt.html#kernel_address

Should be 0x8000 for 32bit, 0x200000 for 64 bit.  (I had neither - 0x80000 by accident - I don't think qemu is too fussy, within a certain range).

The link also states we should be able to specify the address in `config.txt`.  I'll have to try it out on an actual rpi...

Reply:
This is conflating two separate things. Those are _physical_ addresses, where the loader puts the kernel binary in RAM, but the linker config file specifies _virtual_ addresses, assuming that we're running the kernel under virtual memory. The kernel should be linked at whatever virtual addresses it expects to run at.

Now, we haven't turned on paging yet, so for now it's ok to link the kernel at a physical address, but it won't run there long-term.